docs/proposals/dramatically-simplify-cluster-creation.md

```diff
@@ -0,0 +1,222 @@
+
+* *Upgrade*: Later, when Kubernetes 1.4.1 or any newer release is published, I can upgrade to it by typing one other command on each computer.
+
+* *HA*: If one of the computers in the cluster fails, the cluster carries on working. I can find out how to replace the failed computer, including if the computer was one of the masters.
+
+## Top-down view: UX for Phase I items
+
+We will introduce a new binary, kubeadm, which ships with the Kubernetes OS packages (and binary tarballs, for OSes without package managers).
+
+```
+laptop$ kubeadm --help
+
+kubeadm: bootstrap a kubernetes cluster.
+
+Global options:
+
+    --discovery    Discovery mechanism to use. Choose from: gossip,
+                   out-of-band. Default: gossip.
+
+Subcommands:
+    init master    Run this on the first server you deploy onto.
+
+    join node      Run this on other servers to join an existing cluster.
+
+    user create    Create user and fetch authentication config for them.
+```
+
+### Install
+
+*On first machine:*
+
+```
+master# kubeadm init master
+Initializing kubernetes master...  [done]
+Cluster token: 73R2SIPM739TNZOA
+Run the following command on machines you want to become nodes:
+  kubeadm join node --token=73R2SIPM739TNZOA <master-ip>
+You can now run kubectl here.
+```
+
+*On N "node" machines:*
+
+```
+node# kubeadm join node --token=73R2SIPM739TNZOA <master-ip>
+Initializing kubernetes node...    [done]
+Bootstrapping certificates...      [done]
+Joined node to cluster, see 'kubectl get nodes' on master.
+```
+
+Note `[done]` would be colored green in all of the above.
+
+### Install: alternative for automated deploy
+
+*The user (or their config management system) creates a token and passes the same one to both init and join.*
+
+```
+master# kubeadm init master --token=73R2SIPM739TNZOA
+Initializing kubernetes master...  [done]
+You can now run kubectl here.
+```
+
+### Pre-flight check
+
+```
+master# kubeadm init master
+Error: socat not installed. Unable to proceed.
+```
+
+### Control
+
+*On master, after Install, kubectl is automatically able to talk to localhost:8080:*
+
+```
+master# kubectl get pods
+[normal kubectl output]
+```
+
+*To mint new user credentials on the master:*
+
+```
+master# kubeadm user create -o kubeconfig-bob bob
+
+Waiting for cluster to become ready...       [done]
+Creating user certificate for user...        [done]
+Waiting for user certificate to be signed... [done]
+Your cluster configuration file has been saved in kubeconfig.
+
+laptop# scp <master-ip>:/root/kubeconfig-bob ~/.kubeconfig
+laptop# kubectl get pods
+[normal kubectl output]
+```
+
+### Install-addons
+
+*Using CNI network as example:*
+
+```
+master# kubectl apply --purge -f \
+    https://git.io/kubernetes-addons/<X>.yaml
+[normal kubectl apply output]
+```
+
+### Add-node
+
+*Same as Install – "on node machines".*
+
+### Secure
+
+```
+node# kubeadm join --token=GARBAGE node <master-ip>
+Unable to join mesh network. Check your token.
```

Comment:
OK, I'll change this to "Unable to authenticate to <master-ip>. Check your token."